manifests/ssl_bump.pp

```diff
@@ -1,41 +1,29 @@
-# @summary 
+# @summary
 #   Defines ssl_bump entries for a squid server.
 # @see
-#   http://www.squid-cache.org/Doc/config/ssl_bump/ 
+#   http://www.squid-cache.org/Doc/config/ssl_bump/
 # @example
 #   squid::ssl_bump { 'all':
 #     action => 'bump',
 #   }
-# 
+#
 #   Adds a squid.conf line
 #   ssl_bump bump all
-# 
-# @param title 
+#
+# @param title
 #   The name of acl the ssl_bump rule is applied to
-# @param action 
+# @param action
 #   The type of the ssl_bump, must be defined, e.g bump, peek, ..
 # @param order
 #   Order can be used to configure where in `squid.conf`this configuration section should occur.
 define squid::ssl_bump (
-  Enum[
-    'bump',
-    'client-first',
-    'none',
-    'peek',
-    'peek-and-splice',
-    'server-first',
-    'splice',
-    'stare',
-    'terminate']
-          $action = 'bump',
-  String  $value  = $title,
-  String  $order  = '05',
+  Squid::Action::SslBump $action = 'bump',
```

Comment:
This is the only type I've added.  If anyone has the energy, they can add other types in a separate PR.